# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptSettings.cpp
# Commit: 121bf804d2a6
# Full Hash: 121bf804d2a615ddccd0f5d8ed20e40d4715ff08
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-25 09:30:56
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Add the setup for WebTaskSchedulingState r=jjaschke,dom-core,arai
#   
#   Spec: https://wicg.github.io/scheduling-apis/#scheduling-state
#   
#   This patch add the setup for WebTaskSchedulingState so that it
# ==============================================================================

diff -r 6b23965ce57e -r 121bf804d2a6 dom/script/ScriptSettings.cpp
--- a/dom/script/ScriptSettings.cpp	Mon Feb 24 19:31:26 2025 +0000
+++ b/dom/script/ScriptSettings.cpp	Mon Feb 24 19:31:26 2025 +0000
@@ -233,6 +233,13 @@
   return xpc::NativeGlobal(global);
 }
 
+WebTaskSchedulingState* GetWebTaskSchedulingState() {
+  if (const nsIGlobalObject* global = GetEntryGlobal()) {
+    return global->GetWebTaskSchedulingState();
+  }
+  return nullptr;
+}
+
 nsIPrincipal* GetWebIDLCallerPrincipal() {
   MOZ_ASSERT(NS_IsMainThread());
   ScriptSettingsStackEntry* entry = ScriptSettingsStack::EntryPoint();